src/prefect/orion/orchestration/rules.py

```diff
@@ -237,11 +238,12 @@ async def validate_proposed_state(
             await self._validate_proposed_state()
             return
         except Exception as exc:
-            logger.exception("Encountered error during state validation")
+            error_key = hashlib.md5(str(exc).encode())
+            error_msg = f"Internal server error while validating state ({error_key}"
```

Comment:
Possibly a dumb question, but what are we doing with this error key?

Reply:
The user could provide the key to a server admin to lookup the internal error.